File: fix_jsx.py
import re

path = '/app/src/pages/RunPipeline.jsx'
with open(path, 'r', encoding='utf-8', errors='replace') as f:
    content = f.read()

# Fix broken line: useState(false) missing semicolon + stray );
fixed = re.sub(
    r'(const \[optionalOpen, setOptionalOpen\] = useState\(false\))\s*\n[\s\S]*?\);',
    r'\1;',
    content,
    count=1
)

if content != fixed:
    with open(path, 'w', encoding='utf-8') as f:
        f.write(fixed)
    print('FIXED: Repaired broken optionalOpen useState line')
else:
    # Show context to debug
    lines = content.split('\n')
    for i, line in enumerate(lines[410:420], start=411):
        print(f'{i}: {repr(line)}')
    print('Pattern not found - showing raw lines above')

File: fix_jsx.js
const fs = require('fs');
const path = '/app/src/pages/RunPipeline.jsx';
let content = fs.readFileSync(path, 'utf-8');

// Show 200 chars around optionalOpen for debug
const idx = content.indexOf('optionalOpen, setOptionalOpen');
const chunk = content.slice(idx - 5, idx + 150);
console.log('BEFORE:', JSON.stringify(chunk));

// Fix: replace broken useState(false) [missing semicolon + stray );] with correct version
// The corruption looks like: useState(false)\n\n\n                );
const fixed = content.replace(
    /const \[optionalOpen, setOptionalOpen\] = useState\(false\)[\s\S]{1,50}?\);/,
    'const [optionalOpen, setOptionalOpen] = useState(false);'
);

if (fixed !== content) {
    fs.writeFileSync(path, fixed, 'utf-8');
    console.log('SUCCESS: Fixed broken optionalOpen line');
} else {
    console.log('No fix applied. Raw context above shows what was found.');
}
